assets/dev/js/editor/document/hooks/ui/settings/reload-preview.js

```diff
@@ -0,0 +1,35 @@
+import After from 'elementor-api/modules/hooks/ui/after';
+
+export class ReloadPreview extends After {
+	getCommand() {
+		return 'document/elements/settings';
+	}
+
+	getId() {
+		return 'save-layout';
+	}
+
+	getContainerType() {
+		return 'document';
+	}
+
+	getConditions( args ) {
+		return !! args.settings.template;
+	}
+
+	apply() {
+		// Logic moved from js assets/dev/js/editor/components/settings/page/manager.js
+		return $e.run( 'document/save/auto', { force: true } )
+			.then( () => {
+				elementor.reloadPreview();
+
+				elementor.once( 'preview:loaded', () => {
+					// The command 'attach-preview' that set the default route called after the event 'preview:loaded',
+					// for that reason the function wait 50 ms and then sets page-settings route.
+					setTimeout( () => {
+						$e.route( 'panel/page-settings/settings' );
+					}, 50 );
```

Comment:
This is not a good practice. We should find another way to save the previous state.

Reply:
Suggestion?

Reply:
You can either move the `document:loaded` event to the end of the `attach-preview` command or, alternatively, save the panel state to a variable and use it once the preview is loaded.

Reply:
1. Try to move the `document:loaded` event to the end of `attach-preview` command.
2. You can save a state what the panel is open and get back there when the preview is reloaded